sql/chunk.sql

```diff
@@ -67,3 +67,10 @@ RETURNS TABLE(chunk_id INTEGER, hypertable_id INTEGER, att_num INTEGER, nullfrac
 slot1numbers FLOAT4[], slot2numbers FLOAT4[], slot3numbers FLOAT4[], slot4numbers FLOAT4[], slot5numbers FLOAT4[],
 slotvaluetypetrings CSTRING[], slot1values CSTRING[], slot2values CSTRING[], slot3values CSTRING[], slot4values CSTRING[], slot5values CSTRING[])
 AS '@MODULE_PATHNAME@', 'ts_chunk_get_colstats' LANGUAGE C VOLATILE;
+
+CREATE OR REPLACE FUNCTION _timescaledb_internal.create_chunk_table(
```

Comment:
We are doing all of this in the context of `chunk copy/move`. So all these new APIs should reflect that. IMO, the name of this function should be `create_chunk_replica` and it needs to take the destination data node name as an input argument. The other argument that's needed is the `chunk_id`.  We should get all the dimension slices info for this chunk.

Then we should call something similar to `chunk_api_create_on_data_nodes` passing this target data node as input. I think we should modify CHUNK_CREATE_STMT to take in an additional argument to not create hypertable/metadata linkages on the DN.

IIUC, isn't the above enough to do what we want here?

Reply:
> We are doing all of this in the context of `chunk copy/move`. So all these new APIs should reflect that. IMO, the name of this function should be `create_chunk_replica` and it needs to take the destination data node name as an input argument. The other argument that's needed is the `chunk_id`. We should get all the dimension slices info for this chunk.

This is a local function, which needs to exist to be able to implement the function on the access node. It is separate task to create the function on the access node, which will take distributed chunk as input. So it is outside the scope of this PR.



> I think we should modify CHUNK_CREATE_STMT/chunk_create to take in an additional argument to not create hypertable/metadata linkages on the DN. You have already made good changes in `chunk_create_from_hypercube_after_lock` and other functions to support this.

This requires to change the signature of `create_chunk` and it brings different semantics to it. Such change requires a team discussion. Also the refactoring of underlying C function will make it more complicated. So imho I am not in favor of this idea.



> Here the assumption is that the hypertable exists on this DN. If it does not then we would need to create that using `attach_data_node` as part of copy/move chunk.

It is outside the scope of this PR.

Reply:
> This is a local function, which needs to exist to be able to implement the function on the access node. It is separate task to create the function on the access node, which will take distributed chunk as input. So it is outside the scope of this PR.
> 

Ok, I see you have a separate work item for the distributed function call.

> > I think we should modify CHUNK_CREATE_STMT/chunk_create to take in an additional argument to not create hypertable/metadata linkages on the DN. You have already made good changes in `chunk_create_from_hypercube_after_lock` and other functions to support this.
> 
> This requires to change the signature of `create_chunk` and it brings different semantics to it. Such change requires a team discussion. Also the refactoring of underlying C function will make it more complicated. So imho I am not in favor of this idea.
> 

Ok. Looks like we should discuss this because your function is basically a cut down version of `create_chunk` AIUI. We could always default this additional argument (if we choose to modify create_chunk) to true. And I don't see this`create_chunk` function in the documentation, so, I don't think it's supposed to be used by users directly anyways.

> > Here the assumption is that the hypertable exists on this DN. If it does not then we would need to create that using `attach_data_node` as part of copy/move chunk.
> 
> It is outside the scope of this PR.

Sure, if not already added, we should add a work item for this in the main copy/move chunk issue.

Reply:
> Looks like we should discuss this because your function is basically a cut down version of `create_chunk` AIUI. We could always default this additional argument (if we choose to modify create_chunk) to true. And I don't see this`create_chunk` function in the documentation, so, I don't think it's supposed to be used by users directly anyways.

It is not just cut down. It also behaves differently if the chunk already exists. So few considerations:
1. The existing `chunk_table` either creates a chunk or returns the existing chunk. `create_chunk_table` either creates the table or fails.
2. The existing `chunk_table` returns `chunk_id`, while `create_chunk_table` will never return `chunk_id`, since it is not allocated at this step and will be allocated during attaching the chunk table.
3. Attaching chunk table will also use a part of `chunk_table` implementation. So if `create_chunk_table` is extended with creating an empty chunk table, then it is logical to include attaching chunk into the same API. IMHO it will make implementation of `chunk_table` more difficult to follow.

The primary usage of `create_chunk_table` is to insert data into a distributed hypertable.

I suggest to implement separate functions first and have a separate discussion about the API, since the functions needed for copy/move chunk might be exposed  as public API in future.